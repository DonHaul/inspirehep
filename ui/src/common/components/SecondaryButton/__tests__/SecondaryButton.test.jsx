import { render } from '@testing-library/react';

import SecondaryButton from '../SecondaryButton';

describe('SecondaryButton', () => {
  it('renders button', () => {
    const { asFragment } = render(
      <SecondaryButton onClick={jest.fn()}>Test</SecondaryButton>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('calls onClick when button is clicked', () => {
    const onClick = jest.fn();
    const { getByRole } = render(
      <SecondaryButton onClick={onClick}>Test</SecondaryButton>
    );
    getByRole('button').click();
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
