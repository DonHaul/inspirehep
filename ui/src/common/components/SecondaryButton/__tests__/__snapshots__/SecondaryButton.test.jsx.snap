// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondaryButton renders button 1`] = `
<DocumentFragment>
  <button
    class="__SecondaryButton__"
    type="button"
  >
    Test
  </button>
</DocumentFragment>
`;
